src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -561,6 +600,34 @@ protected override Symbol MakePatternVariable(DeclarationPatternSyntax node, Syn
                 _containingFieldOpt, nodeToBind);
         }
 
+        protected override Symbol MakePatternVariable(DeconstructionPatternSyntax node, SyntaxNode nodeToBind)
+        {
+            var designation = node.Designation as SingleVariableDesignationSyntax;
+            if (designation == null)
+            {
+                return null;
+            }
+
+            return GlobalExpressionVariable.Create(
+                _containingType, _modifiers, node.Type,
+                designation.Identifier.ValueText, designation, designation.GetLocation(),
+                _containingFieldOpt, nodeToBind);
+        }
+
+        protected override Symbol MakePatternVariable(PropertyPatternSyntax node, SyntaxNode nodeToBind)
```

Comment:
>MakePatternVariable [](start = 34, length = 19)

It looks like we could share some code among three MakePatternVariable overloads.

Reply:
Noted in [https://github.com/dotnet/roslyn/issues/23222](https://github.com/dotnet/roslyn/issues/23222).

---
In reply to: [156511120](https://github.com/dotnet/roslyn/pull/23209#discussion_r156511120) [](ancestors = 156511120)